src/org/exist/webstart/JnlpJarFiles.java

```diff
@@ -129,11 +144,7 @@ public JnlpJarFiles(final JnlpHelper jnlpHelper) {
      * @return list of jar files.
      */
     public List<Path> getAllWebstartJars() {
-        final List<Path> corejars = new ArrayList<>();
-
-        allFiles.values().stream().filter((file) -> (FileUtils.fileName(file).endsWith(".jar"))).forEach(corejars::add);
-
-        return corejars;
+        return allFiles.values().stream().filter((file) -> (FileUtils.fileName(file).endsWith(".jar"))).collect(Collectors.toList());
```

Comment:
This will leak file handles. If you use a `Stream<Path>` then you need to close it. Consider:

```java
try(final Stream<Path> paths = allFiles.values().stream()) {
  return paths.filter((file) -> (FileUtils.fileName(file).endsWith(".jar"))).collect(Collectors.toList());
}
```